mobile-academy/src/test/java/org/motechproject/nms/mobileacademy/it/MobileAcademyServiceBundleIT.java

```diff
@@ -122,6 +123,69 @@ public void testSetExistingBookmark() {
     }
 
     @Test
+    public void testSetLastBookmark() {
+        bookmarkDataService.deleteAll();
+        MaBookmark bookmark = new MaBookmark(9876543210L, 666L, null, null);
+        maService.setBookmark(bookmark);
+        List<Bookmark> added = bookmarkDataService.findBookmarksForUser("9876543210");
+        assertTrue(added.size() == 1);
+
+        bookmark.setBookmark("Chapter11_Quiz");
+        Map<String, Integer> scores = new HashMap<>();
+        for (int i = 1; i < 12; i++) {
+            scores.put(String.valueOf(i), ((int) (Math.random() * 100)) % 5);
+        }
+        bookmark.setScoresByChapter(scores);
+        maService.setBookmark(bookmark);
+    }
+
+    @Test
+    public void testSetGetLastBookmark() {
+        bookmarkDataService.deleteAll();
+        MaBookmark bookmark = new MaBookmark(9987654321L, 666L, null, null);
+        maService.setBookmark(bookmark);
+        List<Bookmark> added = bookmarkDataService.findBookmarksForUser("9987654321");
+        assertTrue(added.size() == 1);
+
+        bookmark.setBookmark("Chapter11_Quiz");
+        Map<String, Integer> scores = new HashMap<>();
+        for (int i = 1; i < 12; i++) {
+            scores.put(String.valueOf(i), 1);
+        }
+        bookmark.setScoresByChapter(scores);
+        maService.setBookmark(bookmark);
+
+        MaBookmark retrieved = maService.getBookmark(9987654321L, 666L);
+        assertNotNull(retrieved.getCallingNumber());
+        assertNotNull(retrieved.getCallId());
+        assertNotNull(retrieved.getBookmark());
+        assertNotNull(retrieved.getScoresByChapter());
+    }
+
+    @Test
+    public void testSetGetResetBookmark() {
```

Comment:
Ah, NVM, I just got it. You set passing scores and check that the bookmark is reset, right?